codegen/module.go

```diff
@@ -284,18 +298,6 @@ func (system *ModuleSystem) ResolveModules(
 		if err != nil {
 			return nil, err
 		}
-
-		// Resolve dependencies again in case two instances
```

Comment:
Resolving twice could in theory be necessary because the first resolve mutates an array that the second resolve reads. Aka you need to keep re-resolving until the array stops growing...

Unless the resolve runs in a deterministic order where it never grows with a second call.